File: Hard/2009. Minimum Number of Operations to Make Array Continuous/2009. Minimum Number of Operations to Make Array Continuous.java
class Solution {
    public int minOperations(int[] nums) {
        int n = nums.length;
        int minOperations = n;
        Set<Integer> uniqueNums = new HashSet<>();
        List<Integer> sortedUniqueNums = new ArrayList<>();
        for (int num : nums) {
            uniqueNums.add(num);
        }
        sortedUniqueNums.addAll(uniqueNums);
        Collections.sort(sortedUniqueNums);

        int right = 0;

        for (int left = 0; left < sortedUniqueNums.size(); left++) {
            while (right < sortedUniqueNums.size() && sortedUniqueNums.get(right) < sortedUniqueNums.get(left) + n) {
                right++;
            }

            minOperations = Math.min(minOperations, n - (right - left));
        }

        return minOperations;
    }
}

File: Hard/2009. Minimum Number of Operations to Make Array Continuous/2009. Minimum Number of Operations to Make Array Continuous.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        length = len(nums)
        min_operations = length
        unique_nums = set(nums)
        sorted_unique_nums = sorted(list(unique_nums))
        right = 0

        for left in range(len(sorted_unique_nums)):
            while (
                right < len(sorted_unique_nums)
                and sorted_unique_nums[right]
                < sorted_unique_nums[left] + length
            ):
                right += 1

            min_operations = min(min_operations, length - (right - left))

        return min_operations
